src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -729,11 +711,23 @@ bool KeyCondition::canConstantBeWrappedByMonotonicFunctions(
 bool KeyCondition::canConstantBeWrappedByFunctions(
     const ASTPtr & ast, size_t & out_key_column_num, DataTypePtr & out_key_column_type, Field & out_value, DataTypePtr & out_type)
 {
-    // Constant expr should use alias names if any
-    String passed_expr_name = ast->getColumnNameWithoutAlias();
-    String expr_name;
-    if (!canConstantBeWrapped(ast, passed_expr_name, expr_name))
-        return false;
+    String expr_name = ast->getColumnNameWithoutAlias();
+
+    if (key_subexpr_names.count(expr_name) == 0)
+    {
+        /// Let's check another one case.
+        /// If our storage was created with moduloLegacy in partition key,
+        /// We can assume that `modulo(...) = const` is the same as `moduloLegacy(...) = const`.
+        /// Replace modulo to moduloLegacy in AST and check if we also have such a column.
```

Comment:
May be add a comment that this check is deliberately not added for `canConstantBeWrappedByMonotonicFunctions`?
(Because what if someone will use such comparisons in partition key after all)

Reply:
yes, reasonable)